tests/testenvironment.proj

```diff
@@ -0,0 +1,133 @@
+    <TestEnvironment Include="jitstress2_jitstressregs0x1000" JitStress="2" JitStressRegs="0x1000" />
+    <TestEnvironment Include="tailcallstress" TailcallStress="1" />
+    <TestEnvironment Include="jitsse2only" EnableAVX="0" EnableSSE3_4="0" />
+    <TestEnvironment Include="jitnosimd" FeatureSIMD="0" />
+    <TestEnvironment Include="jitincompletehwintrinsic" EnableIncompleteISAClass="1" />
+    <!-- testing the legacy SSE encoding -->
+    <TestEnvironment Include="jitx86hwintrinsicnoavx" EnableIncompleteISAClass="1" EnableAVX="0" />
+    <!-- testing SNB/IVB -->
+    <TestEnvironment Include="jitx86hwintrinsicnoavx2" EnableIncompleteISAClass="1" EnableAVX2="0" />
+    <!-- match "jitnosimd", may need to remove after decoupling HW intrinsic from FeatureSIMD -->
+    <TestEnvironment Include="jitx86hwintrinsicnosimd" EnableIncompleteISAClass="1" FeatureSIMD="0" />
+    <TestEnvironment Include="jitnox86hwintrinsic" EnableIncompleteISAClass="1" EnableSSE="0" EnableSSE2="0" EnableSSE3="0" EnableSSSE3="0" EnableSSE41="0" EnableSSE42="0" EnableAVX="0" EnableAVX2="0" EnableAES="0" EnableBMI1="0" EnableBMI2="0" EnableFMA="0" EnableLZCNT="0" EnablePCLMULQDQ="0" EnablePOPCNT="0" />
+    <TestEnvironment Include="gcstress0x3" GCStress="0x3" />
+    <TestEnvironment Include="gcstress0xc" GCStress="0xC" />
+    <TestEnvironment Include="zapdisable" ZapDisable="1" ReadyToRun="0" />
+    <TestEnvironment Include="heapverify1" HeapVerify="1" />
+    <TestEnvironment Include="gcstress0xc_zapdisable" GCStress="0xC" ZapDisable="1" ReadyToRun="0" />
+    <TestEnvironment Include="gcstress0xc_zapdisable_jitstress2" GCStress="0xC" ZapDisable="1" ReadyToRun="0" JitStress="2" />
+    <TestEnvironment Include="gcstress0xc_zapdisable_heapverify1" GCStress="0xC" ZapDisable="1" ReadyToRun="0" HeapVerify="1" />
+    <TestEnvironment Include="gcstress0xc_jitstress1" GCStress="0xC" JitStress="1" />
+    <TestEnvironment Include="gcstress0xc_jitstress2" GCStress="0xC" JitStress="2" />
+    <TestEnvironment Include="gcstress0xc_minopts_heapverify1" GCStress="0xC" JITMinOpts="1" HeapVerify="1" />
+  </ItemGroup>
+
+  <Target Name="OutputCOMPlusVariables" Inputs="@(COMPlusVariable)" Outputs="%(COMPlusVariable.Identity)">
+    <PropertyGroup>
+      <_COMPlusVariable>%(COMPlusVariable.Identity)</_COMPlusVariable>
+      <_COMPlusVariableMetadataName>$(_COMPlusVariable.Replace('COMPlus_', ''))</_COMPlusVariableMetadataName>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <_TestEnvironment Include="@(TestEnvironment)" Condition="'%(Identity)' == '$(Scenario)'" />
+      <_TestEnvironmentWithMetadata Include="@(_TestEnvironment->HasMetadata($(_COMPlusVariableMetadataName)))" />
+
+      <_COMPlusVariable Include="$(_COMPlusVariable)" Condition="@(_TestEnvironmentWithMetadata->Count()) == 1">
+        <Value>@(_TestEnvironmentWithMetadata->Metadata($(_COMPlusVariableMetadataName)))</Value>
+      </_COMPlusVariable>
```

Comment:
I still find this a bit confusing  - it would be nice to comment that this is extracting the specified scenario/testenv's metadata (because msbuild doesn't make that easy), by essentially doing:
```
foreach complusvariable:
  foreach testenv == scenario:
    append this testenv's complusvariable key/value
```
Impressive use of msbuild :)